### DISCUSSION TITLE
Does Livewire have bugs with long Model Primary Keys?

### DISCUSSION THREAD

**Discussion by eyounelson** [livewire/livewire#3061] [7. Bugs (archived)]
### Description

I had set up my model to use MYSQL's [uuid_short()](https://dev.mysql.com/doc/refman/8.0/en/miscellaneous-functions.html#function_uuid-short) for my primary keys. I do that via an eloquent `creating` event.
Currently I have a row with an ID of `99147636904296461` and I have set up route model binding for a full page Livewire component.
I discover that when I try to make an update to my component's property (any property) via `wire:model="xyz"`  I get an exception of this sort:
`Livewire\Exceptions\CorruptComponentPayloadException
Livewire encountered corrupt data when trying to hydrate the [customer.orders.show-order] component. Ensure that the [name, id, data] of the Livewire component wasn't tampered with between requests. `.

I was curious so I took a look at the request being sent to the server by livewire's JS:

```json
{
	"fingerprint": {
		"id": "BeESBPCEpg584Fmo2w9W",
		"locale": "en",
		"method": "GET",
		"name": "customer.orders.show-order",
		"path": "orders/99147636904296461"
	},
	"serverMemo": {
		"checksum": "801f26fc64f8cff389f7fa6101430230b7b049dd51d8b3d3d60a14df893b9d82",
		"children": [],
		"data": {
			"newInstallment": "",
			"order": []
		},
		"dataMeta": {
			"models": {
				"order": {
					"class": "App\\Models\\Order",
					"connection": "mysql",
					"id": 99147636904296460,
					"relations": [
						"status",
						"products"
					]
				}
			}
		},
		"errors": [],
		"htmlHash": "184edb44"
	},
	"updates": [
		{
			"payload": {
				"name": "newInstallment",
				"value": "2"
			},
			"type": "syncInput"
		}
	]
}
```

Note that the `data.order` property is an empty array.

If I manually update my model's ID to a shorter integer eg 122223 this error goes away.

### Exact steps to reproduce
* Create a full page livewire component and set up implicit route model binding
* Set the ID/PK of your model to a large integer eg `99147636904296461`
* Try to update a property of your component, example using `wire:model="prpName"` in your component's blade

### Stripped-down, copy-pastable code snippets

Here is the exact content of my component's class:

```php
class ShowOrder extends Component
{
    public Order $order;

    public string $newInstallment = '';
    
    public function render()
    {
        return view('livewire.customer.orders.show-order')
            ->extends('layouts.app');
    }
}
```

**Context**
 Livewire version: [ 2.4.0]
- Laravel version: [8.31.0]
- Alpine version:  N/A
- Browser: Tested on  Chrome, FireFox

**Comment by austenc** [comment]
Hello @emmannl,

There are problems with integer-based IDs that are larger than javascript's maximum "safe" integer value. 

This has been brought up before, but unfortunately there isn't a built-in solution for it at this time. Thank you for bringing it up again so that we know other folks are also having this problem!

See here for more information if you want:
https://github.com/livewire/livewire/pull/1938

**Comment by calebporzio** [comment]
I tried reopening #1938, bit can't because the fork is gone.

I'm good with a new PR as long as someone runs benchmarks to make sure this doesn't add any tangible performance overhead.

Also, I'm ok with the traits implementation, but not how is scattered around the codebase like he mentions. I believe there is a more central place to put it.

**Comment by seabasss** [comment]
Did anyone come up with a solution for this? Snowflake ID's is a no-go out of the box with livewire as of now unfortunately. Thank you!

**Reply by eyounelson** [reply]
You can [tty this out](https://github.com/kra8/laravel-snowflake#javascript-support)

**Reply by seabasss** [reply]
Thanks, I’m not using that snowflake package though. Would that just make the integers shorter? Think the best would be if livewire could auto-cast long ints as strings. Without knowing what other consequences that would involve :)

**Reply by eyounelson** [reply]
It will make the intergers shorter (53-bit  which Javascript supports)
